File: java-engine/src/main/java/engine/algorithms/faultless_dfs_dispersion/models/FaultlessDfsDispersionEdge.java
package engine.algorithms.faultless_dfs_dispersion.models;

import engine.core.models.Edge;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(callSuper = false)
public class FaultlessDfsDispersionEdge extends Edge {

    private int fromPort;
    private int toPort;

}

File: dispersion-app/src/app/models/algorithms/faultless-dfs-dispersion/FaultlessDfsDispersionEdge.ts
import { Edge } from '../../core/Edge';

export class FaultlessDfsDispersionEdge extends Edge {
  fromPort: number;
  toPort: number;

  constructor(
    id: number,
    fromID: number,
    toID: number,
    fromPort: number,
    toPort: number
  ) {
    super(id, fromID, toID);
    this.fromPort = fromPort;
    this.toPort = toPort;
  }
}
